cpp/include/cudf/copying.hpp

```diff
@@ -211,7 +211,7 @@ std::unique_ptr<table> scatter(
 /**
  * @brief Indicates when to allocate a mask, based on an existing mask.
  */
-enum class mask_allocation_policy {
+enum class mask_allocation_policy : int32_t {
```

Comment:
why not a smaller type?

Reply:
No particular reason other than that's what our other enums use. I think everything has probably just copied `type_id`: https://github.com/rapidsai/cudf/blob/branch-24.02/cpp/include/cudf/types.hpp#L187